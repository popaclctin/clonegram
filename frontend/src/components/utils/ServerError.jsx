import React from 'react';
import './ServerError.style.scss';

function ServerError({ error }) {
  let content = error.data.message;
  if (error.data.message === 'VALIDATION_ERROR') {
    content = (
      <ul>
        {error.data.invalidParams.map((param, index) => (
          <li key={index}>{param.msg}</li>
        ))}
      </ul>
    );
  }
  return <div className='serverError'>{content}</div>;
}

export default ServerError;
