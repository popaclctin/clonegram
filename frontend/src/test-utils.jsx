import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { setupStore } from './store';
import { setupListeners } from '@reduxjs/toolkit/dist/query';

import { BrowserRouter } from 'react-router-dom';

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  } = {}
) {
  setupListeners(store.dispatch);

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
