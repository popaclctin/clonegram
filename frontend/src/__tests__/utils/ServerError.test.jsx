import React from 'react';
import ServerError from '../../components/utils/ServerError';
import { renderWithProviders } from '../../test-utils';

describe('<ServerError />', () => {
  it('renders correctly', () => {
    const error = { data: { message: 'This is a test' } };
    const { asFragment } = renderWithProviders(<ServerError error={error} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
