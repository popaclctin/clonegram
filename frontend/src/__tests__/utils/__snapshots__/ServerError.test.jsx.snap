// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServerError /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="serverError"
  >
    This is a test
  </div>
</DocumentFragment>
`;
